src/Main.java

```diff
@@ -1,278 +1 @@
-package com.company;
 
-import javax.swing.text.StyledEditorKit;
-import java.util.Scanner;
-import java.io.File;
-import java.io.*;
-import java.util.Queue;
-import java.util.*;
-
-class Cell{
-    int x;
-    int y;
-    int gCost;
-    int hCost;
-    Cell parent;
-
-    public Cell(int x, int y, Cell parent){
-        this.x = x;
-        this.y = y;
-        this.parent = parent;
-    }
-
-    public Cell(int x, int y, Cell parent, int gCost, int hCost){
-        this.x = x;
-        this.y = y;
-        this.parent = parent;
-        this.gCost = gCost;
-        this.hCost = hCost;
-    }
-
-    public Cell getParent(){
-        return this.parent;
-    }
-}
-
-class MazeSolver {
-
-    //4 possible moves
-    static int[] move_x = {1, -1, 0, 0};
-    static int[] move_y = {0, 0, 1, -1};
-
-    // Using DFS algorithm, does not guarantee shortest path
-    static Boolean MazeBackTracker(char[][] maze, Cell cell, int m, int n){
-        // Boundary conditions
-        if (!isValidMove(maze, cell, m, n)) return false;
-
-        // Found exit
-        if (maze[cell.x][cell.y] == 'E') return true;
-
-        // Mark current node as visited
-        maze[cell.x][cell.y] = 'X';
-
-        //Recursively visit the next nodes
-        for (int move = 0; move < 4; move++){
-            Cell nextCell = new Cell(cell.x + move_x[move], cell.y + move_y[move], cell);
-            if (MazeBackTracker(maze, nextCell, m, n)) return true;
```

Comment:
Yeah, so this just guarantees a single path because of the short circuit here (i.e. the early breaking of the loop). I think that this is not a correct solution. I know that the "README" additionally asks "can you find the shortest path", but I read that as 'you should make it find the shortest path', especially since the difference between the solution which finds the shortest path and the one that finds any path isn't that gigantic.